### PR TITLE
Add S.SM.ExtensionCollection to public contract.

### PR DESCRIPTION
**Pull request by StephenBonikowsky**
* Already used by WCF internally, no tests need to be explicitly written for it.

### CODE REVIEW

**Comment by mconnew** [comment]
`InstanceContext` and `OperationContext` both have an `Extensions` property which uses this class. We should probably expose them in the contract.

**Comment by StephenBonikowsky** [comment]
@dotnet-bot test outerloop Ubuntu14.04

**Comment by StephenBonikowsky** [comment]
@dotnet-bot test outerloop Ubuntu14.04

**Comment by StephenBonikowsky** [comment]
@dotnet-bot test outerloop Ubuntu14.04

**Comment by StephenBonikowsky** [comment]
@mconnew @hongdai @zhenlan can you review when you get a chance?

File: src/System.ServiceModel.Primitives/tests/ServiceModel/ExtensionCollection.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            _ownerAttached = owner;
+        }
+
+        public void Detach(IMyExtensibleObject owner)
+        {
+            _ownerDetached = owner;
+        }
+    }
+
+    public class MyExtension2 : IMyExtension
```
Comment (KKhurin) [inline src/System.ServiceModel.Primitives/tests/ServiceModel/ExtensionCollection.cs]: Inherit MyExtension2 from MyExtension1 and avoid code dup in test? :)

Comment (StephenBonikowsky) [inline src/System.ServiceModel.Primitives/tests/ServiceModel/ExtensionCollection.cs]: Good idea!

**Comment by StephenBonikowsky** [comment]
@KKhurin could I get an LGTM please?

**Comment by KKhurin** [comment]
LGTM

**Pull request merged**